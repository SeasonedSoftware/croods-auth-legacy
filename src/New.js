import React from 'react'
import { New } from 'croods'

import Created from './Created'

export default props => {
  const {
    name,
    path,
    method = 'post',
    currentUser,
    setCurrentUser,
    defaultParse,
    render,
    renderCreated,
    ...otherProps
  } = props
  return (
    <New
      name={name}
      path={path}
      method={method}
      parseResponse={
        defaultParse ? undefined : ({ data }) => ({ created: data })
      }
      render={({ create, creating, error }) => {
        return render({
          create,
          creating,
          error,
          currentUser,
          setCurrentUser,
          ...otherProps,
        })
      }}
      renderCreated={created => (
        <Created {...props} render={renderCreated} user={created} />
      )}
    />
  )
}
