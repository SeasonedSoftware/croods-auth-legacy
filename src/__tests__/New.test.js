import React from 'react'
import renderer from 'react-test-renderer'

import New from '../New'

jest.mock('croods', () => ({
  New: props => (
    <div>
      New - {props.render(props)} - {props.renderCreated({ foo: 'bar' })}
    </div>
  ),
}))

jest.mock('../Created', () => props => (
  <div {...props}>Created - {props.renderCreated({ foo: 'bar' })}</div>
))

const props = {
  name: 'auth.Test',
  path: '/auth/test',
  render: jest.fn(() => null),
  renderCreated: jest.fn(() => null),
  foo: 'bar',
}

it('renders correctly', () => {
  const tree = renderer.create(<New {...props} />).toJSON()
  expect(props.render).toBeCalled()
  expect(props.renderCreated).toBeCalled()
  expect(tree).toMatchSnapshot()
})

describe('with different method', () => {
  const tree = renderer.create(<New {...props} method="put" />).toJSON()
  expect(props.render).toBeCalled()
  expect(props.renderCreated).toBeCalled()
  expect(tree).toMatchSnapshot()
})
