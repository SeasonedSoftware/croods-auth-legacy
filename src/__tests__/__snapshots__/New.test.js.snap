// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  New - 
   - 
  <div
    foo="bar"
    method="put"
    name="auth.Test"
    path="/auth/test"
    render={[Function]}
    renderCreated={[Function]}
    user={
      Object {
        "foo": "bar",
      }
    }
  >
    Created - 
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  New - 
   - 
  <div
    foo="bar"
    name="auth.Test"
    path="/auth/test"
    render={[Function]}
    renderCreated={[Function]}
    user={
      Object {
        "foo": "bar",
      }
    }
  >
    Created - 
  </div>
</div>
`;
